Class inheriting from abstract record complains about unimplemented <Clone>$

**Version Used**: 3.8.0-3.20421.1 ([9f0b1046](https://github.com/dotnet/roslyn/commit/9f0b1046))

**Steps to Reproduce**:

Encountered while refactoring code to take advantage of records: ([SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAN0QSwGYE8A+EwsCEAxjAAQJxkD2CAJpQILEykUBKtDzA3gF8AsACgxAAQBMlAMp0wcAMIAbaFEohW7TjB70mlIWKA===))

```csharp
abstract record AbstractRecord {}

class SomeClass : AbstractRecord {}
```

**Expected Behavior**:

A lone CS8865 error:

```
error CS8865: Only records may inherit from records.
```

**Actual Behavior**:

You get an additional red herring error about not implementing `<Clone>$`:

```
error CS8865: Only records may inherit from records.
error CS0534: 'SomeClass' does not implement inherited abstract member 'AbstractRecord.<Clone>$()'
```

---------------------

Not the worst thing ever, but it does leak the unspeakable synthesized method, which could be confusing to developers who aren't aware of the internal mechanics of records.

Additionally, due to how squiggles for these errors are shown Visual Studio, the red herring can be the first one the developer sees:

![image](https://user-images.githubusercontent.com/278957/90954730-ad024800-e43c-11ea-8c0c-e172e4a3147d.png)

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs
@@ -83,7 +83,7 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
             foreach (var abstractMember in this.AbstractMembers)
             {
                 // Dev10 reports failure to implement properties/events in terms of the accessors
-                if (abstractMember.Kind == SymbolKind.Method)
+                if (abstractMember.Kind == SymbolKind.Method && abstractMember is not SynthesizedRecordOrdinaryMethod)
                 {
                     diagnostics.Add(ErrorCode.ERR_UnimplementedAbstractMethod, this.Locations[0], this, abstractMember);
                 }

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs
@@ -1307,7 +1307,7 @@ class SomeClass : AbstractRecord {}";
             comp.VerifyDiagnostics(
                 // (3,19): error CS8865: Only records may inherit from records.
                 // class SomeClass : AbstractRecord {}
-                Diagnostic(ErrorCode.ERR_BadInheritanceFromRecord, "AbstractRecord").WithLocation(5, 19)
+                Diagnostic(ErrorCode.ERR_BadInheritanceFromRecord, "AbstractRecord").WithLocation(3, 19)
             );
         }
 

--- a/src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs
+++ b/src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs
@@ -1295,6 +1295,22 @@ data struct S2(int X, int Y);";
             );
         }
 
+        [WorkItem(44781, "https://github.com/dotnet/roslyn/issues/44781")]
+        [Fact]
+        public void ClassInheritingFromRecord()
+        {
+            var src = @"
+abstract record AbstractRecord {}
+class SomeClass : AbstractRecord {}";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (3,19): error CS8865: Only records may inherit from records.
+                // class SomeClass : AbstractRecord {}
+                Diagnostic(ErrorCode.ERR_BadInheritanceFromRecord, "AbstractRecord").WithLocation(5, 19)
+            );
+        }
+
         [Fact]
         public void RecordInheritance()
         {

